src/Extensions/Activities/Internal/ActivityListenerInitializerService.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Omex.Extensions.Abstractions.Activities.Processing;
+
+namespace Microsoft.Omex.Extensions.Activities
+{
+	internal class ActivityListenerInitializerService : IHostedService
+	{
+		private readonly IActivityStartObserver[] m_activityStartObservers;
+		private readonly IActivityStopObserver[] m_activityStopObservers;
+		private readonly IActivityListenerConfigurator m_configurator;
+		private ActivityListener? m_listener;
+
+		public ActivityListenerInitializerService(
+			IEnumerable<IActivityStartObserver> activityStartObservers,
+			IEnumerable<IActivityStopObserver> activityStopObservers,
+			IActivityListenerConfigurator configurator)
+		{
+			m_activityStartObservers = activityStartObservers.ToArray();
+			m_activityStopObservers = activityStopObservers.ToArray();
+			m_configurator = configurator;
+		}
+
+		public Task StartAsync(CancellationToken cancellationToken)
+		{
+			m_listener = new ActivityListener();
+			m_listener.ActivityStarted += ActivityStart;
+			m_listener.ActivityStopped += ActivityStopped;
+			m_listener.Sample += Sample;
+			m_listener.SampleUsingParentId += SampleUsingParentId;
+			m_listener.ShouldListenTo += ShouldListenTo;
+
+			ActivitySource.AddActivityListener(m_listener);
+
+			return Task.CompletedTask;
+		}
+
+		public Task StopAsync(CancellationToken cancellationToken)
```

Comment:
>public [](start = 2, length = 6)

Do we have comments for public methods in internal classes?

Reply:
Internal classes don't contribute to documentations, so it makes sense to add comments only when you need to clarify things for somebody who already see the code. In this case it would be `<inheridoc/>` anyways since the method coming from interface